// @walletconnect/modal-react-native/lib/module/controllers/ApiCtrl.js
import { Platform } from 'react-native';
import { subscribeKey as subKey } from 'valtio/utils';
import { proxy } from 'valtio/vanilla';
import { CoreHelperUtil } from '../utils/CoreHelperUtil';
import { FetchUtil } from '../utils/FetchUtil';
import { StorageUtil } from '../utils/StorageUtil';
import { ConfigCtrl } from './ConfigCtrl';
import { AssetCtrl } from './AssetCtrl';
import { SDK_VERSION } from '../constants/Config';

// -- Helpers ------------------------------------------- //
const baseUrl = CoreHelperUtil.getApiUrl();
const api = new FetchUtil({
  baseUrl
});
const defaultEntries = '48';
const recommendedEntries = '8';
const sdkType = 'wcm';

// -- Types --------------------------------------------- //

// -- State --------------------------------------------- //
const state = proxy({
  sdkVersion: `modal-react-native-${SDK_VERSION}`,
  page: 1,
  count: 0,
  recommended: [],
  wallets: [],
  search: [],
  installed: []
});

// -- Controller ---------------------------------------- //
export const ApiCtrl = {
  state,
  platform() {
    return Platform.select({
      default: 'ios',
      android: 'android'
    });
  },
  subscribeKey(key, callback) {
    return subKey(state, key, callback);
  },
  setSdkVersion(sdkVersion) {
    state.sdkVersion = sdkVersion;
  },
  _getApiHeaders() {
    return {
      'x-project-id': ConfigCtrl.state.projectId,
      'x-sdk-type': sdkType,
      'x-sdk-version': state.sdkVersion
    };
  },
  async _fetchWalletImage(imageId) {
    const imageUrl = `${api.baseUrl}/getWalletImage/${imageId}`;
    AssetCtrl.setWalletImage(imageId, imageUrl);
  },
  async fetchInstalledWallets() {
    const path = Platform.select({
      default: 'getIosData',
      android: 'getAndroidData'
    });
    const {
      data: walletData
    } = await api.get({
      path,
      headers: ApiCtrl._getApiHeaders()
    });
    const promises = walletData.map(async item => {
      return {
        id: item.id,
        isInstalled: await CoreHelperUtil.checkInstalled(item)
      };
    });
    const results = await Promise.all(promises);
    const installed = results.filter(_ref => {
      let {
        isInstalled
      } = _ref;
      return isInstalled;
    }).map(_ref2 => {
      let {
        id
      } = _ref2;
      return id;
    });
    const {
      explorerExcludedWalletIds
    } = ConfigCtrl.state;
    const excludeWalletIds = CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    if (installed.length > 0) {
      const {
        data
      } = await api.get({
        path: '/getWallets',
        headers: ApiCtrl._getApiHeaders(),
        params: {
          page: '1',
          platform: this.platform(),
          entries: installed === null || installed === void 0 ? void 0 : installed.length.toString(),
          include: installed === null || installed === void 0 ? void 0 : installed.join(','),
          exclude: excludeWalletIds === null || excludeWalletIds === void 0 ? void 0 : excludeWalletIds.join(',')
        }
      });
      const walletImages = data.map(d => d.image_id).filter(Boolean);
      await Promise.allSettled(walletImages.map(id => ApiCtrl._fetchWalletImage(id)));
      state.installed = data;
    }
  },
  async fetchRecommendedWallets() {
    const {
      installed
    } = ApiCtrl.state;
    const {
      explorerRecommendedWalletIds,
      explorerExcludedWalletIds
    } = ConfigCtrl.state;
    const excludeWalletIds = CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    const includeWalletIds = CoreHelperUtil.isArray(explorerRecommendedWalletIds) ? explorerRecommendedWalletIds : [];
    const exclude = [...installed.map(_ref3 => {
      let {
        id
      } = _ref3;
      return id;
    }), ...(excludeWalletIds ?? [])].filter(Boolean);
    const {
      data,
      count
    } = await api.get({
      path: '/getWallets',
      headers: ApiCtrl._getApiHeaders(),
      params: {
        page: '1',
        platform: this.platform(),
        entries: recommendedEntries,
        include: includeWalletIds === null || includeWalletIds === void 0 ? void 0 : includeWalletIds.join(','),
        exclude: exclude === null || exclude === void 0 ? void 0 : exclude.join(',')
      }
    });
    const recent = await StorageUtil.getRecentWallet();
    const recommendedImages = data.map(d => d.image_id).filter(Boolean);
    await Promise.allSettled([...recommendedImages, recent ? recent.image_id : undefined].map(id => ApiCtrl._fetchWalletImage(id)));
    state.recommended = data;
    state.count = count ?? 0;
  },
  async fetchWallets(_ref4) {
    let {
      page
    } = _ref4;
    const {
      explorerExcludedWalletIds
    } = ConfigCtrl.state;
    const excludedIds = CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    const exclude = [...state.installed.map(_ref5 => {
      let {
        id
      } = _ref5;
      return id;
    }), ...state.recommended.map(_ref6 => {
      let {
        id
      } = _ref6;
      return id;
    }), ...(excludedIds ?? [])].filter(Boolean);
    const {
      data,
      count
    } = await api.get({
      path: '/getWallets',
      headers: ApiCtrl._getApiHeaders(),
      params: {
        page: String(page),
        platform: this.platform(),
        entries: String(defaultEntries),
        exclude: exclude.join(',')
      }
    });
    const images = data.map(w => w.image_id).filter(Boolean);
    await Promise.allSettled([...images.map(id => ApiCtrl._fetchWalletImage(id)), CoreHelperUtil.wait(300)]);
    state.wallets = [...state.wallets, ...data];
    state.count = count > state.count ? count : state.count;
    state.page = page;
  },
  async searchWallet(_ref7) {
    let {
      search
    } = _ref7;
    const {
      explorerExcludedWalletIds
    } = ConfigCtrl.state;
    const excludeWalletIds = CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    state.search = [];
    const {
      data
    } = await api.get({
      path: '/getWallets',
      headers: ApiCtrl._getApiHeaders(),
      params: {
        page: '1',
        platform: this.platform(),
        entries: String(defaultEntries),
        search,
        exclude: excludeWalletIds === null || excludeWalletIds === void 0 ? void 0 : excludeWalletIds.join(',')
      }
    });
    const images = data.map(w => w.image_id).filter(Boolean);
    await Promise.allSettled([...images.map(id => ApiCtrl._fetchWalletImage(id)), CoreHelperUtil.wait(300)]);
    state.search = data;
  },
  async prefetch() {
    await ApiCtrl.fetchInstalledWallets();
    state.prefetchPromise = Promise.race([Promise.allSettled([ApiCtrl.fetchRecommendedWallets()]), CoreHelperUtil.wait(3000)]);
  }
};
