// @walletconnect/modal-react-native/lib/commonjs/components/Shimmer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Shimmer = void 0;
var _reactNativeSvg = require("react-native-svg");
var _reactNative = require("react-native");
var _useTheme = _interopRequireDefault(require("../hooks/useTheme"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const AnimatedRect = _reactNative.Animated.createAnimatedComponent(_reactNativeSvg.Rect);
const Shimmer = _ref => {
  let {
    width = 200,
    height = 200,
    duration = 1000,
    borderRadius = 0,
    backgroundColor,
    foregroundColor,
    style
  } = _ref;
  const animatedValue = new _reactNative.Animated.Value(0);
  const Theme = (0, _useTheme.default)();
  const animatedProps = {
    fill: animatedValue.interpolate({
      inputRange: [0, 0.5, 1],
      outputRange: [backgroundColor ?? Theme.background2, foregroundColor ?? Theme.background3, backgroundColor ?? Theme.background3]
    }),
    width,
    height,
    x: 0,
    y: 0,
    rx: borderRadius,
    ry: borderRadius
  };
  _reactNative.Animated.loop(_reactNative.Animated.timing(animatedValue, {
    toValue: 1,
    duration,
    useNativeDriver: false
  })).start();
  return /*#__PURE__*/React.createElement(_reactNativeSvg.Svg, {
    width: width,
    height: height,
    style: style
  }, /*#__PURE__*/React.createElement(AnimatedRect, animatedProps));
};
exports.Shimmer = Shimmer;
