// @walletconnect/modal-react-native/lib/commonjs/components/WalletItemLoader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WalletItemLoader = WalletItemLoader;
exports.WalletItemLoaderHeight = void 0;
var _reactNative = require("react-native");
var _Shimmer = require("./Shimmer");
var _WalletItem = require("./WalletItem");
const WalletItemLoaderHeight = 98;
exports.WalletItemLoaderHeight = WalletItemLoaderHeight;
function WalletItemLoader(_ref) {
  let {
    style
  } = _ref;
  return /*#__PURE__*/React.createElement(_reactNative.View, {
    style: [styles.container, style]
  }, /*#__PURE__*/React.createElement(_Shimmer.Shimmer, {
    height: 60,
    width: 60,
    borderRadius: 17
  }), /*#__PURE__*/React.createElement(_Shimmer.Shimmer, {
    height: 14,
    width: 52,
    borderRadius: 6,
    style: styles.text
  }));
}
const styles = _reactNative.StyleSheet.create({
  container: {
    height: _WalletItem.WALLET_HEIGHT,
    alignItems: 'center',
    justifyContent: 'center',
    width: _WalletItem.WALLET_WIDTH,
    borderRadius: 16
  },
  text: {
    marginTop: 8
  }
});
