// @walletconnect/modal-react-native/lib/commonjs/components/WalletItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WalletItem = exports.WALLET_WIDTH = exports.WALLET_MARGIN = exports.WALLET_HEIGHT = exports.WALLET_FULL_HEIGHT = void 0;
var _react = require("react");
var _reactNative = require("react-native");
var _useTheme = _interopRequireDefault(require("../hooks/useTheme"));
var _UiUtil = require("../utils/UiUtil");
var _Touchable = _interopRequireDefault(require("./Touchable"));
var _WalletImage = _interopRequireDefault(require("./WalletImage"));
var _valtio = require("valtio");
var _ApiCtrl = require("../controllers/ApiCtrl");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const WALLET_MARGIN = 8;
exports.WALLET_MARGIN = WALLET_MARGIN;
const WALLET_WIDTH = 80;
exports.WALLET_WIDTH = WALLET_WIDTH;
const WALLET_HEIGHT = 98;
exports.WALLET_HEIGHT = WALLET_HEIGHT;
const WALLET_FULL_HEIGHT = WALLET_HEIGHT + WALLET_MARGIN * 2;
exports.WALLET_FULL_HEIGHT = WALLET_FULL_HEIGHT;
function _WalletItem(_ref) {
  let {
    id,
    name,
    imageUrl,
    style,
    isRecent,
    onPress
  } = _ref;
  const Theme = (0, _useTheme.default)();
  const imageHeaders = _ApiCtrl.ApiCtrl._getApiHeaders();
  const {
    installed
  } = (0, _valtio.useSnapshot)(_ApiCtrl.ApiCtrl.state);
  const isInstalled = !!(installed !== null && installed !== void 0 && installed.find(wallet => wallet.id === id));
  return /*#__PURE__*/React.createElement(_Touchable.default, {
    onPress: onPress,
    key: id,
    style: [styles.container, style]
  }, /*#__PURE__*/React.createElement(_WalletImage.default, {
    size: "md",
    url: imageUrl,
    imageHeaders: imageHeaders
  }), /*#__PURE__*/React.createElement(_reactNative.Text, {
    style: [styles.name, {
      color: Theme.foreground1
    }],
    numberOfLines: 1
  }, _UiUtil.UiUtil.getWalletName(name, true)), (isRecent || isInstalled) && /*#__PURE__*/React.createElement(_reactNative.Text, {
    style: [styles.status, {
      color: Theme.foreground3
    }]
  }, isRecent ? 'RECENT' : 'INSTALLED'));
}
const styles = _reactNative.StyleSheet.create({
  container: {
    width: WALLET_WIDTH,
    height: WALLET_HEIGHT,
    alignItems: 'center',
    marginVertical: WALLET_MARGIN
  },
  name: {
    marginTop: 5,
    maxWidth: 100,
    fontSize: 12,
    fontWeight: '600',
    textAlign: 'center',
    marginBottom: 2
  },
  status: {
    fontSize: 10,
    fontWeight: '700',
    textAlign: 'center'
  }
});
const WalletItem = /*#__PURE__*/(0, _react.memo)(_WalletItem, (prevProps, nextProps) => {
  return prevProps.name === nextProps.name;
});
exports.WalletItem = WalletItem;
var _default = WalletItem;
exports.default = _default;
