// @walletconnect/modal-react-native/lib/module/partials/wcm-all-wallets-list/index.js
import { View, FlatList } from 'react-native';
import { useSnapshot } from 'valtio';
import useTheme from '../../hooks/useTheme';
import WalletItem, { WALLET_FULL_HEIGHT, WALLET_MARGIN } from '../../components/WalletItem';
import { AssetUtil } from '../../utils/AssetUtil';
import { ConfigCtrl } from '../../controllers/ConfigCtrl';
import styles from './styles';
import { WalletItemLoader } from '../../components/WalletItemLoader';
export function AllWalletsList(_ref) {
  let {
    columns,
    itemWidth,
    windowHeight,
    themeMode,
    isPortrait,
    onWalletPress,
    loading,
    onFetchNextPage,
    list
  } = _ref;
  const {
    recentWallet
  } = useSnapshot(ConfigCtrl.state);
  const Theme = useTheme();
  const loadingTemplate = items => {
    return /*#__PURE__*/React.createElement(View, {
      style: [styles.loaderContainer, {
        height: Math.round(windowHeight * 0.6)
      }]
    }, Array.from({
      length: items
    }).map((_, index) => /*#__PURE__*/React.createElement(WalletItemLoader, {
      key: index,
      style: {
        marginBottom: WALLET_MARGIN * 2,
        width: itemWidth
      }
    })));
  };
  const renderWallet = _ref2 => {
    let {
      item
    } = _ref2;
    return /*#__PURE__*/React.createElement(WalletItem, {
      id: item.id,
      name: item.name,
      isRecent: item.id === (recentWallet === null || recentWallet === void 0 ? void 0 : recentWallet.id),
      onPress: () => onWalletPress(item),
      imageUrl: AssetUtil.getWalletImage(item),
      style: {
        width: itemWidth
      }
    });
  };
  if (loading) {
    return loadingTemplate(20);
  }
  return /*#__PURE__*/React.createElement(FlatList, {
    data: list,
    style: {
      height: Math.round(windowHeight * 0.6),
      backgroundColor: Theme.background1
    },
    contentContainerStyle: styles.listContentContainer,
    indicatorStyle: themeMode === 'dark' ? 'white' : 'black',
    showsVerticalScrollIndicator: true,
    numColumns: columns,
    fadingEdgeLength: 20,
    onEndReached: onFetchNextPage,
    onEndReachedThreshold: 2,
    key: isPortrait ? 'portrait' : 'landscape',
    getItemLayout: (_data, index) => ({
      length: WALLET_FULL_HEIGHT,
      offset: WALLET_FULL_HEIGHT * index,
      index
    }),
    renderItem: renderWallet
  });
}
