// @walletconnect/modal-react-native/lib/module/utils/FetchUtil.js
// -- Types ----------------------------------------------------------------------

// -- Utility --------------------------------------------------------------------
export class FetchUtil {
  constructor(_ref) {
    let {
      baseUrl
    } = _ref;
    this.baseUrl = baseUrl;
  }
  async get(_ref2) {
    let {
      headers,
      ...args
    } = _ref2;
    const url = this.createUrl(args).toString();
    const response = await fetch(url, {
      method: 'GET',
      headers
    });
    return response.json();
  }
  async post(_ref3) {
    let {
      body,
      headers,
      ...args
    } = _ref3;
    const url = this.createUrl(args).toString();
    const response = await fetch(url, {
      method: 'POST',
      headers,
      body: body ? JSON.stringify(body) : undefined
    });
    return response.json();
  }
  async put(_ref4) {
    let {
      body,
      headers,
      ...args
    } = _ref4;
    const url = this.createUrl(args).toString();
    const response = await fetch(url, {
      method: 'PUT',
      headers,
      body: body ? JSON.stringify(body) : undefined
    });
    return response.json();
  }
  async delete(_ref5) {
    let {
      body,
      headers,
      ...args
    } = _ref5;
    const url = this.createUrl(args).toString();
    const response = await fetch(url, {
      method: 'DELETE',
      headers,
      body: body ? JSON.stringify(body) : undefined
    });
    return response.json();
  }
  createUrl(_ref6) {
    let {
      path,
      params
    } = _ref6;
    const url = new URL(path, this.baseUrl);
    if (params) {
      Object.entries(params).forEach(_ref7 => {
        let [key, value] = _ref7;
        if (value) {
          url.searchParams.append(key, value);
        }
      });
    }
    return url;
  }
}
