// @walletconnect/modal-react-native/lib/commonjs/modal/wcm-modal-router/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalRouter = ModalRouter;
var _react = require("react");
var _valtio = require("valtio");
var _useOrientation = require("../../hooks/useOrientation");
var _wcmQrCodeView = _interopRequireDefault(require("../../views/wcm-qr-code-view"));
var _wcmAllWalletsView = _interopRequireDefault(require("../../views/wcm-all-wallets-view"));
var _RouterCtrl = require("../../controllers/RouterCtrl");
var _wcmConnectView = _interopRequireDefault(require("../../views/wcm-connect-view"));
var _wcmConnectingView = _interopRequireDefault(require("../../views/wcm-connecting-view"));
var _useTheme = _interopRequireDefault(require("../../hooks/useTheme"));
var _reactNative = require("react-native");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function ModalRouter(props) {
  const routerState = (0, _valtio.useSnapshot)(_RouterCtrl.RouterCtrl.state);
  const Theme = (0, _useTheme.default)();
  const {
    height,
    width,
    isPortrait
  } = (0, _useOrientation.useOrientation)();
  const ViewComponent = (0, _react.useMemo)(() => {
    switch (routerState.view) {
      case 'ConnectWallet':
        return _wcmConnectView.default;
      case 'WalletExplorer':
        return _wcmAllWalletsView.default;
      case 'Qrcode':
        return _wcmQrCodeView.default;
      case 'Connecting':
        return _wcmConnectingView.default;
      default:
        return _wcmConnectView.default;
    }
  }, [routerState.view]);
  return /*#__PURE__*/React.createElement(_reactNative.View, {
    style: [styles.wrapper, {
      backgroundColor: Theme.background1
    }]
  }, /*#__PURE__*/React.createElement(ViewComponent, _extends({
    windowHeight: height,
    windowWidth: width,
    isPortrait: isPortrait
  }, props)));
}
const styles = _reactNative.StyleSheet.create({
  wrapper: {
    borderTopLeftRadius: 30,
    borderTopRightRadius: 30
  }
});
