// @walletconnect/modal-react-native/lib/commonjs/controllers/ApiCtrl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ApiCtrl = void 0;
var _reactNative = require("react-native");
var _utils = require("valtio/utils");
var _vanilla = require("valtio/vanilla");
var _CoreHelperUtil = require("../utils/CoreHelperUtil");
var _FetchUtil = require("../utils/FetchUtil");
var _StorageUtil = require("../utils/StorageUtil");
var _ConfigCtrl = require("./ConfigCtrl");
var _AssetCtrl = require("./AssetCtrl");
var _Config = require("../constants/Config");
// -- Helpers ------------------------------------------- //
const baseUrl = _CoreHelperUtil.CoreHelperUtil.getApiUrl();
const api = new _FetchUtil.FetchUtil({
  baseUrl
});
const defaultEntries = '48';
const recommendedEntries = '8';
const sdkType = 'wcm';

// -- Types --------------------------------------------- //

// -- State --------------------------------------------- //
const state = (0, _vanilla.proxy)({
  sdkVersion: `modal-react-native-${_Config.SDK_VERSION}`,
  page: 1,
  count: 0,
  recommended: [],
  wallets: [],
  search: [],
  installed: []
});

// -- Controller ---------------------------------------- //
const ApiCtrl = {
  state,
  platform() {
    return _reactNative.Platform.select({
      default: 'ios',
      android: 'android'
    });
  },
  subscribeKey(key, callback) {
    return (0, _utils.subscribeKey)(state, key, callback);
  },
  setSdkVersion(sdkVersion) {
    state.sdkVersion = sdkVersion;
  },
  _getApiHeaders() {
    return {
      'x-project-id': _ConfigCtrl.ConfigCtrl.state.projectId,
      'x-sdk-type': sdkType,
      'x-sdk-version': state.sdkVersion
    };
  },
  async _fetchWalletImage(imageId) {
    const imageUrl = `${api.baseUrl}/getWalletImage/${imageId}`;
    _AssetCtrl.AssetCtrl.setWalletImage(imageId, imageUrl);
  },
  async fetchInstalledWallets() {
    const path = _reactNative.Platform.select({
      default: 'getIosData',
      android: 'getAndroidData'
    });
    const {
      data: walletData
    } = await api.get({
      path,
      headers: ApiCtrl._getApiHeaders()
    });
    const promises = walletData.map(async item => {
      return {
        id: item.id,
        isInstalled: await _CoreHelperUtil.CoreHelperUtil.checkInstalled(item)
      };
    });
    const results = await Promise.all(promises);
    const installed = results.filter(_ref => {
      let {
        isInstalled
      } = _ref;
      return isInstalled;
    }).map(_ref2 => {
      let {
        id
      } = _ref2;
      return id;
    });
    const {
      explorerExcludedWalletIds
    } = _ConfigCtrl.ConfigCtrl.state;
    const excludeWalletIds = _CoreHelperUtil.CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    if (installed.length > 0) {
      const {
        data
      } = await api.get({
        path: '/getWallets',
        headers: ApiCtrl._getApiHeaders(),
        params: {
          page: '1',
          platform: this.platform(),
          entries: installed === null || installed === void 0 ? void 0 : installed.length.toString(),
          include: installed === null || installed === void 0 ? void 0 : installed.join(','),
          exclude: excludeWalletIds === null || excludeWalletIds === void 0 ? void 0 : excludeWalletIds.join(',')
        }
      });
      const walletImages = data.map(d => d.image_id).filter(Boolean);
      await Promise.allSettled(walletImages.map(id => ApiCtrl._fetchWalletImage(id)));
      state.installed = data;
    }
  },
  async fetchRecommendedWallets() {
    const {
      installed
    } = ApiCtrl.state;
    const {
      explorerRecommendedWalletIds,
      explorerExcludedWalletIds
    } = _ConfigCtrl.ConfigCtrl.state;
    const excludeWalletIds = _CoreHelperUtil.CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    const includeWalletIds = _CoreHelperUtil.CoreHelperUtil.isArray(explorerRecommendedWalletIds) ? explorerRecommendedWalletIds : [];
    const exclude = [...installed.map(_ref3 => {
      let {
        id
      } = _ref3;
      return id;
    }), ...(excludeWalletIds ?? [])].filter(Boolean);
    const {
      data,
      count
    } = await api.get({
      path: '/getWallets',
      headers: ApiCtrl._getApiHeaders(),
      params: {
        page: '1',
        platform: this.platform(),
        entries: recommendedEntries,
        include: includeWalletIds === null || includeWalletIds === void 0 ? void 0 : includeWalletIds.join(','),
        exclude: exclude === null || exclude === void 0 ? void 0 : exclude.join(',')
      }
    });
    const recent = await _StorageUtil.StorageUtil.getRecentWallet();
    const recommendedImages = data.map(d => d.image_id).filter(Boolean);
    await Promise.allSettled([...recommendedImages, recent ? recent.image_id : undefined].map(id => ApiCtrl._fetchWalletImage(id)));
    state.recommended = data;
    state.count = count ?? 0;
  },
  async fetchWallets(_ref4) {
    let {
      page
    } = _ref4;
    const {
      explorerExcludedWalletIds
    } = _ConfigCtrl.ConfigCtrl.state;
    const excludedIds = _CoreHelperUtil.CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    const exclude = [...state.installed.map(_ref5 => {
      let {
        id
      } = _ref5;
      return id;
    }), ...state.recommended.map(_ref6 => {
      let {
        id
      } = _ref6;
      return id;
    }), ...(excludedIds ?? [])].filter(Boolean);
    const {
      data,
      count
    } = await api.get({
      path: '/getWallets',
      headers: ApiCtrl._getApiHeaders(),
      params: {
        page: String(page),
        platform: this.platform(),
        entries: String(defaultEntries),
        exclude: exclude.join(',')
      }
    });
    const images = data.map(w => w.image_id).filter(Boolean);
    await Promise.allSettled([...images.map(id => ApiCtrl._fetchWalletImage(id)), _CoreHelperUtil.CoreHelperUtil.wait(300)]);
    state.wallets = [...state.wallets, ...data];
    state.count = count > state.count ? count : state.count;
    state.page = page;
  },
  async searchWallet(_ref7) {
    let {
      search
    } = _ref7;
    const {
      explorerExcludedWalletIds
    } = _ConfigCtrl.ConfigCtrl.state;
    const excludeWalletIds = _CoreHelperUtil.CoreHelperUtil.isArray(explorerExcludedWalletIds) ? explorerExcludedWalletIds : [];
    state.search = [];
    const {
      data
    } = await api.get({
      path: '/getWallets',
      headers: ApiCtrl._getApiHeaders(),
      params: {
        page: '1',
        platform: this.platform(),
        entries: String(defaultEntries),
        search,
        exclude: excludeWalletIds === null || excludeWalletIds === void 0 ? void 0 : excludeWalletIds.join(',')
      }
    });
    const images = data.map(w => w.image_id).filter(Boolean);
    await Promise.allSettled([...images.map(id => ApiCtrl._fetchWalletImage(id)), _CoreHelperUtil.CoreHelperUtil.wait(300)]);
    state.search = data;
  },
  async prefetch() {
    await ApiCtrl.fetchInstalledWallets();
    state.prefetchPromise = Promise.race([Promise.allSettled([ApiCtrl.fetchRecommendedWallets()]), _CoreHelperUtil.CoreHelperUtil.wait(3000)]);
  }
};
exports.ApiCtrl = ApiCtrl;
