// @walletconnect/modal-react-native/lib/module/components/Image.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import { useRef } from 'react';
import { Animated, Image as NativeImage, Platform } from 'react-native';
function Image(_ref) {
  let {
    source,
    headers,
    style,
    ...rest
  } = _ref;
  const opacity = useRef(new Animated.Value(0));
  const isIOS = Platform.OS === 'ios';

  // Fade in image on load for iOS. Android does this by default.
  const onLoadEnd = () => {
    Animated.timing(opacity.current, {
      toValue: 1,
      duration: 200,
      useNativeDriver: true
    }).start();
  };
  return isIOS ? /*#__PURE__*/React.createElement(Animated.Image, _extends({
    source: {
      uri: source,
      headers
    },
    onLoadEnd: onLoadEnd,
    style: [{
      opacity: opacity.current
    }, style]
  }, rest)) : /*#__PURE__*/React.createElement(NativeImage, _extends({
    source: {
      uri: source,
      headers
    },
    style: style
  }, rest));
}
export default Image;
