// @walletconnect/modal-react-native/lib/module/components/Shimmer.js
import { Svg, Rect } from 'react-native-svg';
import { Animated } from 'react-native';
import useTheme from '../hooks/useTheme';
const AnimatedRect = Animated.createAnimatedComponent(Rect);
export const Shimmer = _ref => {
  let {
    width = 200,
    height = 200,
    duration = 1000,
    borderRadius = 0,
    backgroundColor,
    foregroundColor,
    style
  } = _ref;
  const animatedValue = new Animated.Value(0);
  const Theme = useTheme();
  const animatedProps = {
    fill: animatedValue.interpolate({
      inputRange: [0, 0.5, 1],
      outputRange: [backgroundColor ?? Theme.background2, foregroundColor ?? Theme.background3, backgroundColor ?? Theme.background3]
    }),
    width,
    height,
    x: 0,
    y: 0,
    rx: borderRadius,
    ry: borderRadius
  };
  Animated.loop(Animated.timing(animatedValue, {
    toValue: 1,
    duration,
    useNativeDriver: false
  })).start();
  return /*#__PURE__*/React.createElement(Svg, {
    width: width,
    height: height,
    style: style
  }, /*#__PURE__*/React.createElement(AnimatedRect, animatedProps));
};
